src/core.ts

```diff
@@ -0,0 +1,24 @@
+import { makeComponent } from './component';
+import { makeContext } from './create-context';
+
+type GenericRenderer = (this: unknown, ...args: any[]) => unknown | void;
+type RenderFunction = (result: unknown, container: DocumentFragment | Element) => void;
+
+function haunted({ render }: { render: RenderFunction }) {
```

Comment:
Can be better to expose options interface, making it easier to enxtend it later, as:

```ts
export interface Options {
  render: RenderFunction;
}

function haunted({ render }: Options)
```

Does it make sense?